internal/products.go

```diff
@@ -58,7 +60,26 @@ func parseProducts(reader io.Reader) ([]Product, error) {
 			loggedError("Can't read product information: %v", err.Error())
 	}
 
-	err = json.Unmarshal(data, &products)
+	dec := json.NewDecoder(bytes.NewBuffer(data))
+	t, err := dec.Token()
+	if err != nil {
+		return products,
+			loggedError("Can't read product information: %v - %s", err.Error(), data)
+	}
+
+	// Depending on which API was used the JSON we get passed contains either
+	// an list for products ("/connect/subscriptions/products") with a single
+	// element in it. Or a single product. So we need to Unmarshall() slightly
+	// different for both cases.
+	if t == json.Delim('[') {
```

Comment:
I would skip this additional verification step and just try to unmarshal into the slice and if this fails into the single product. If both fails, we error out.

Reply:
Makes sense. I'll rework that.

Reply:
Done